### PR TITLE
Add example system

### LINKED ISSUE #86: Add a more complete example

### PR DESCRIPTION
**Pull request by jonlamb-gh**
Adds an example system that runs on
the Boundary Devices SABRE Lite i.MX6 Development Board (sabrelite).
    
Note that it currently does not run on real hardware yet, which requires additional configuration
around IOMUXing and ENET phy initialization.

Closes #86

### CODE REVIEW

File: examples/system/applications/console/src/main.rs [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+        pub fn cmd(
+            _menu: &Menu<Context>,
+            item: &Item<Context>,
+            args: &[&str],
+            context: &mut Context,
+        ) {
+            let addr = menu::argument_finder(item, args, "addr").unwrap().unwrap();
+            let mut addr_octets = [0_u8; 4];
+            for (idx, part) in addr.split('.').into_iter().enumerate() {
+                addr_octets[idx] = part.parse().unwrap();
```
Comment (ZackPierce) [inline examples/system/applications/console/src/main.rs:320]: I suppose the fact that invalid input will cause the system to crash is acceptable for a simple example.

Comment (jonlamb-gh) [reply examples/system/applications/console/src/main.rs:320]: yeah, no application level error handling is done, and it's kinda fun to crash the app and see the rest of the system remain operational.

**Review by ZackPierce — CHANGES_REQUESTED** [review]
A variety of nitpicks and notes, but overall it's extremely impressive.

File: examples/system/libraries/debug-logger/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            Some("debug") => LevelFilter::Debug,
+            Some("trace") => LevelFilter::Trace,
+            _ => LevelFilter::Debug,
+        }
+    }
+}
+
+impl log::Log for DebugLogger {
+    fn enabled(&self, _metadata: &Metadata) -> bool {
+        true
```
[the remark this answers is not in the document]
Comment (jonlamb-gh) [reply examples/system/libraries/debug-logger/src/lib.rs:25]: Probably, I'm just not using `log_enabled!` anywhere in here so it never was issue, I'll get that fixed :+1:

File: examples/system/imx6-hal/src/enet/mod.rs [hunk 3]
```diff
@@ -0,0 +1,416 @@
...
+            mac,
+            rx_ring,
+            tx_ring,
+        })
+    }
+
+    pub fn init(&mut self) {
+        log::trace!("[enet] init");
+
+        // TODO - need to configure IOMUX
```
[the remark this answers is not in the document]
Comment (jonlamb-gh) [reply examples/system/imx6-hal/src/enet/mod.rs:156]: Not currently, will do once we tackle https://github.com/auxoncorp/ferros/issues/88
I'll add notes in the comments linking to that ticket.

**Review by ZackPierce — APPROVED** [review]
:+1:

**Pull request merged**